pkg/etcdcli/etcdcli.go

```
@@ -462,3 +426,57 @@ func (g *etcdClientGetter) Defragment(ctx context.Context, member *etcdserverpb.
 	}
 	return resp, nil
 }
+
+func endpoints(nodeLister corev1listers.NodeLister,
+	configmapsLister corev1listers.ConfigMapLister,
+	networkLister configv1listers.NetworkLister,
+	nodeListerSynced cache.InformerSynced,
+	configmapsListerSynced cache.InformerSynced,
+	networkListerSynced cache.InformerSynced) ([]string, error) {
+
+	if !nodeListerSynced() {
+		return nil, fmt.Errorf("node lister not synced")
+	}
+	if !configmapsListerSynced() {
+		return nil, fmt.Errorf("configmaps lister not synced")
+	}
+	if !networkListerSynced() {
+		return nil, fmt.Errorf("network lister not synced")
+	}
+
+	var etcdEndpoints []string
+	// This configmap should always only contain voting members. Learners should not be present here.
+	configmap, err := configmapsLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
+	if err != nil {
+		klog.Errorf("failed to list endpoints from %s/%s, falling back to listing nodes: %w",
+			operatorclient.TargetNamespace, "etcd-endpoints", err)
+
+		network, err := networkLister.Get("cluster")
+		if err != nil {
+			return nil, fmt.Errorf("failed to list cluster network: %w", err)
+		}
+
+		nodes, err := nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
```

swallowing an error here, fixing...